linkis-computation-governance/linkis-manager/linkis-resource-manager/src/main/scala/org/apache/linkis/resourcemanager/external/yarn/YarnResourceRequester.scala

```diff
@@ -212,8 +213,20 @@ class YarnResourceRequester extends ExternalResourceRequester with Logging {
     httpGet.addHeader("Accept", "application/json")
     if (this.provider.getConfigMap.get("authorEnable").asInstanceOf[Boolean])
       httpGet.addHeader(HttpHeaders.AUTHORIZATION, "Basic " + getAuthorizationStr)
-    val response = YarnResourceRequester.httpClient.execute(httpGet)
-    parse(EntityUtils.toString(response.getEntity()))
+    var httpResponse: HttpResponse = null
+    if(this.provider.getConfigMap.get("kerberosEnable") != null
+      && this.provider.getConfigMap.get("kerberosEnable").asInstanceOf[Boolean]){
+      val principalName = this.provider.getConfigMap.get("principalName").asInstanceOf[String]
```

Comment:
Do you need to consider the case where the parameter is empty?

Reply:
If the parameter is empty, use the else method directly